src/ImageArchive/UiIntegration.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Class UIIntegration.
+ *
+ * @package P4\MasterTheme\ImageArchive
+ */
+
+namespace P4\MasterTheme\ImageArchive;
+
+use P4\MasterTheme\Capability;
+use P4\MasterTheme\Features;
+use P4\MasterTheme\Loader;
+
+/**
+ * Add some WordPress UI elements if the feature is active.
+ */
+class UiIntegration {
+	/**
+	 * UiIntegration constructor.
+	 */
+	public function __construct() {
+		self::hooks();
+	}
+
+	/**
+	 * Hook up to WordPress.
+	 */
+	private static function hooks() {
+		if ( ! Features::active( Features::IMAGE_ARCHIVE ) ) {
+			return;
+		}
+		add_action( 'admin_menu', [ self::class, 'create_admin_menu' ] );
+		add_filter( 'media_upload_tabs', [ self::class, 'image_archive_tab' ] );
+		add_action( 'media_upload_image_archive', [ self::class, 'output_image_picker' ] );
+		add_action( 'post-upload-ui', [ self::class, 'media_library_post_upload_ui' ] );
+	}
+
+	/**
+	 * Add GPI Media Library upload button in WP media popup upload UI.
+	 *
+	 * @todo: This is preserved from the original plugin, but can probably be done in a better way.
+	 */
+	public static function media_library_post_upload_ui() {
+		global $pagenow;
+		$classes = 'button media-button button-primary button-large add_media switchtoml';
+
+		// Add the insert media class only when not in the editor, i.e. when on the "Media > Add New" page.
+		if ( ( 'post.php' !== $pagenow ) && ( ! in_array( get_post_type(), [ 'post', 'page', 'campaign' ], true ) ) ) {
+			$classes .= ' insert-media';
+		}
+		print '<button id="db-upload-btn" class="' . $classes . '">' . esc_html__( //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			'Upload From GPI Media Library',
+			'planet4-master-theme-backend'
+		) . '</button>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * @param string[] $tabs Existing tabs passed by the filter.
+	 *
+	 * @return string[] Same tabs with image archive tab added to them.
+	 */
+	public static function image_archive_tab( $tabs ): array {
+		$tabs['image_archive'] = 'GPI Image Archive';
```

Comment:
will this one be translatable ?

Reply:
Good catch, I did this similarly to how it was done in the media library plugin, but indeed it looks like the translation needs to be performed here :+1: